core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -776,6 +776,9 @@ object SparkSubmit extends CommandLineUtils with Logging {
       sparkConf.set("spark.submit.pyFiles", formattedPyFiles)
     }
 
+    // [SPARK-22744] Cannot get the submit hostname of application
+    sparkConf.set("spark.submit.hostname", Utils.localHostName)
```

Comment:
So basically how do you leverage this configuration?

I think maybe you can achieve this out of Spark, like changing the submit script to dump some information about submission hostname.

It is a little awkward to add a configuration in Spark which only used out of Spark.

Reply:
Agree your suggestion. Thanks @jerryshao. 
In contrast to MR, there is a read-only property mapreduce.job.submithostname. So I thought it should treate as spark internal, not out of. Anyway, your suggestion work around.

Reply:
You could close this PR if you think it's worthless to merge. No opinion.

Reply:
I don't have the permission to close your PR, you should do it yourself.